### PR TITLE
[TFTRT] Check supported precisions before build

### PR DESCRIPTION
**Pull request by nvkevihu**
Checks the platform's supported precisions prior to building each engine. If the selected precision mode is not supported, terminate the program with a detailed message.

cc: @DEKHTIARJonathan @meena-at-work

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -1050,6 +1050,31 @@ Status Converter::Init(nvinfer1::ILogger* trt_logger) {
   VLOG(1) << "Creating TensorRT builder";
   trt_builder_.reset(nvinfer1::createInferBuilder(*trt_logger));
 
+  // Check if platform supports enabled precision mode
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:1053]: Please add a period to the end of the comment.
s/enabled/the/ ?

File: tensorflow/python/compiler/tensorrt/utils.py [hunk 2]
```diff
@@ -257,3 +261,79 @@ def draw_graphdef_as_graphviz(graphdef, dot_output_filename):
   print("  - `sudo apt install -y graphviz`")
   print("  - `dot -Tpng <input_filename>.dot -o <output_filename>.png`")
   print("===================================================================\n")
+
+
+@functools.lru_cache
+def get_nvidia_gpu_desc_dict():
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/utils.py:267]: This routine is not used outside this module, right?
Let's make it internal in this case.

File: tensorflow/python/compiler/tensorrt/utils.py [hunk 3]
```diff
@@ -257,3 +261,79 @@ def draw_graphdef_as_graphviz(graphdef, dot_output_filename):
...
+      raise e
+
+    else:
+      logging.error(str(e))
+      return False
+
+
+@functools.lru_cache
+def is_platform_supported(precision, gpu_id=0):
+  """ This function determines if the current NVIDIA GPU platform supports the
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/utils.py:301]: A docstring should start with a one-line summary, see [here](https://google.github.io/styleguide/pyguide.html#381-docstrings)

File: tensorflow/python/compiler/tensorrt/utils.py [hunk 4]
```diff
@@ -257,3 +261,79 @@ def draw_graphdef_as_graphviz(graphdef, dot_output_filename):
...
+  requested TensorRT compute precision.
+
+  Required Compute Capabilities:
+  - FP16 = 5.3 || 6.0 || 6.2 || 7.0+
+  - INT8 = 6.1 || 7.0 || 7.2+
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/utils.py:306]: Please also document the argument and the exceptions.

**Review by bixia1 — APPROVED** [review]

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 5]
```diff
@@ -1050,6 +1050,31 @@ Status Converter::Init(nvinfer1::ILogger* trt_logger) {
...
+      "\nFor more information, please refer to: ",
+      "https://docs.nvidia.com/deeplearning/tensorrt/support-matrix/index.html#hardware-precision-matrix",
+      "\n",
+      "\nRequired Compute Capabilities:",
+      "\nINT8 = 6.1 || 7.0 || 7.2+",
+      "\nFP16 = 5.3 || 6.0 || 6.2 || 7.0+",
+      "\n",
+      "\nPlease reconvert & rebuild your model with a supported precision."
+    );
+    LOG(FATAL) << error_message;
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:1075]: The submit system doesn't like LOG(FATAL) anymore. It is suggested to LOG(ERROR) and then return an error status.

**Review by bixia1 — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@nvkevihu Can you please check @bixia1's comments and resolve conflicts?. Thank you!

**Comment by gbaned** [comment]
@nvkevihu Can you please resolve conflicts? Thank you!

**Comment by nvkevihu** [comment]
Sorry for the delay, resolved conflicts and addressed comments.

**Review by bixia1 — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @bixia1 Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @bixia1 Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @poulsbo Can you please review this PR ? Thank you!

**Comment by poulsbo** [comment]
@pjannaty
Hi Pooya, could you please review (or mark as reviewed by NVIDIA) this PR? If I review it here in Github, then I will need to a find a separate reviewer on the internal Google side, which will slow things down. It's best if I do the reviews on the Google side, and NVIDIA staff (yourself or Nathan representing NVIDIA) handle the Github side.

**Review by pjannaty — DISMISSED** [review]
Approved based on multiple reviews and approvals by Bixia!

**Comment by poulsbo** [comment]
I'm seeing a lot of leaks, apparently from `createInferBuilder_INTERNAL`, when testing on GPU.

@pjannaty does this repro on your side?

```
[  SKIPPED ] 4 tests, listed below:
[  SKIPPED ] ConverterTest.MaybeApplyQuantizationRanges
[  SKIPPED ] OpConvTestInstantiation/OpConverter_FP32_Test.ConvertQDQDynamicRangeMode/0
[  SKIPPED ] OpConvTestInstantiation/OpConverter_FP32_Test.ConvertQDQDynamicRangeMode/1
[  SKIPPED ] OpConvTestInstantiation/OpConverter_FP32_Test.ConvertQDQDynamicRangeMode/2
E0303 15:15:43.119625    9797 heap-profile-table.cc:410] RAW: Leak check _main_ detected leaks of 696520 bytes in 694 objects
E0303 15:15:43.120497    9797 heap-profile-table.cc:430] RAW: The 20 largest leaks:
E0303 15:15:43.215939    9797 heap-profile-table.cc:451] RAW: Leak of 28800 bytes in 9 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b64589 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.216113    9797 heap-profile-table.cc:451] RAW: Leak of 28800 bytes in 9 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b5b0a9 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.216294    9797 heap-profile-table.cc:451] RAW: Leak of 28800 bytes in 9 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b5afc9 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.216508    9797 heap-profile-table.cc:451] RAW: Leak of 28800 bytes in 9 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b64f49 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.216651    9797 heap-profile-table.cc:451] RAW: Leak of 28800 bytes in 9 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b63c49 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.216799    9797 heap-profile-table.cc:451] RAW: Leak of 28800 bytes in 9 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b65109 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.216942    9797 heap-profile-table.cc:451] RAW: Leak of 28800 bytes in 9 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b651e9 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.217087    9797 heap-profile-table.cc:451] RAW: Leak of 25600 bytes in 8 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b64e69 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.217229    9797 heap-profile-table.cc:451] RAW: Leak of 25600 bytes in 8 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b63b69 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.217433    9797 heap-profile-table.cc:451] RAW: Leak of 19200 bytes in 6 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b65029 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.217716    9797 heap-profile-table.cc:451] RAW: Leak of 19200 bytes in 6 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b5d3c9 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.217885    9797 heap-profile-table.cc:451] RAW: Leak of 19200 bytes in 6 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b65569 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.218061    9797 heap-profile-table.cc:451] RAW: Leak of 19200 bytes in 6 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b59c49 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.218221    9797 heap-profile-table.cc:451] RAW: Leak of 19200 bytes in 6 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b65489 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.218413    9797 heap-profile-table.cc:451] RAW: Leak of 19200 bytes in 6 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b59609 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.218602    9797 heap-profile-table.cc:451] RAW: Leak of 16000 bytes in 5 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b5c1e9 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.218768    9797 heap-profile-table.cc:451] RAW: Leak of 16000 bytes in 5 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b5ae09 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.218924    9797 heap-profile-table.cc:451] RAW: Leak of 16000 bytes in 5 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b5a529 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.219065    9797 heap-profile-table.cc:451] RAW: Leak of 16000 bytes in 5 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b5a369 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start

E0303 15:15:43.219211    9797 heap-profile-table.cc:451] RAW: Leak of 16000 bytes in 5 objects allocated from:
	@ 0x7f98b085dbe6 NewHook()
	@ 0x7f9897278828 absl::base_internal::MallocHook::InvokeNewHookSlow()
	@ 0x556381bfd2ca slow_alloc<>()
	@ 0x556374640260 createInferBuilder_INTERNAL
	@ 0x7f9b77b58677 tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::ParameterizedOpConverterTestBase()
	@ 0x7f9b77b653a9 testing::internal::ParameterizedTestFactory<>::CreateTest()
	@ 0x7f9b5b9a65d8 testing::TestInfo::Run()
	@ 0x7f9b5b9a74e5 testing::TestSuite::Run()
	@ 0x7f9b5b9b8214 testing::internal::UnitTestImpl::RunAllTests()
	@ 0x7f9b5b9b7241 testing::UnitTest::Run()
	@ 0x7f9b77b4b9a1 main
	@ 0x7f98930bd633 __libc_start_main
	@ 0x55637425d5ea [../sysdeps/x86_64/start.S:122](https://cs.corp.google.com/piper///depot/sysdeps/x86_64/start.S?l=122&ws=tap-presubmit-server/19104818&snapshot=2) _start
```

**Comment by pjannaty** [comment]
@DEKHTIARJonathan who is most familiar with this work and can check the leaks Alan mentioned above?

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan / @pjannaty Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @nvkevihu, @DEKHTIARJonathan, @pjannaty  I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen! Thank you!

**Pull request closed without merging**